ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx

```diff
@@ -79,16 +124,23 @@ const OperationAmount = ({ operation }: OperationComponentProps) => {
 };
 
 const generateTimelineOperationListItemText = (operation: OperationListDTO) => {
-  const operationTitle =
-    "maskedPan" in operation
-      ? I18n.t(
-          `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.${operation.operationType}`,
-          { maskedPan: operation.maskedPan }
-        )
-      : I18n.t(
-          `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.${operation.operationType}`
-        );
+  const operationTitle = () => {
+    if ("channel" in operation && operation.channel === ChannelEnum.QRCODE) {
+      return I18n.t(
+        "idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.QRCODE_TRANSACTION"
+      );
+    }
 
+    if ("maskedPan" in operation) {
+      return I18n.t(
+        `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.${operation.operationType}`,
+        { maskedPan: operation.maskedPan }
+      );
+    }
+    return I18n.t(
+      `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.${operation.operationType}`
+    );
```

Comment:
while it does indeed look cleaner, this solution as is breaks in case of an operation that requires `maskedPan` and is neither a transaction nor a reversal (see payment method activation, for example).
So far, the actual solution works as intended and is (in my humble opinion, of course) pretty readable, but will most likely need a refactor with the style you're suggesting as soon as new evolutions come through.

```ts
  const operationTitle = () => {
    switch (operation.operationType) {
      case TransactionOperationTypeEnum.TRANSACTION:
        if (operation.channel === ChannelEnum.QRCODE) {
          return I18n.t(
            "idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.QRCODE_TRANSACTION"
          );
        }
      // eslint-disable-next-line no-fallthrough
      case InstrumentOperationTypeEnum.ADD_INSTRUMENT:
        return I18n.t(
          `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.${operation.operationType}`,
          { maskedPan: operation.maskedPan }
        );
      default:
        return I18n.t(
          `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.${operation.operationType}`
        );
    }
 ```
 maybe something like this? after all, channel appears to be mandatory only in case of a `TRANSACTION` operation. 
 (and also there is no design for a `REVERSAL` operation with `QRCODE` channel yet ).
 Lastly, even if TypedI18n filters it out silently, to me it seems a bit un-idiomatic to pass `maskedPan` to an I18n key that does not require it, like `TRANSACTION`; what do you think?

Reply:
Yes you are right.
But in that case you need to include all types of operation which requires `maskedPan` :
- `RejectedInstrumentOperationDTO`: `REJECTED_ADD_INSTRUMENT`, `REJECTED_DELETE_INSTRUMENT`
- `TransactionOperationDTO`: `TRANSACTION`, `REVERSAL`
- `InstrumentOperationDTO`: `ADD_INSTRUMENT`, `DELETE_INSTRUMENT`

Reply:
That's right. I was looking at DTOs and not at the UI 🙄